pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -1028,8 +1031,21 @@ func isGatewayMatch(gateway string, gatewayNames []string) bool {
 	return false
 }
 
-func buildGatewayVirtualHostDomains(hostname string, port int) []string {
+func buildGatewayVirtualHostDomains(service *model.Service, hostname string, port int) []string {
 	domains := []string{hostname}
+	if service != nil && service.Attributes.ServiceRegistry == provider.Kubernetes {
```

Comment:
I think the `service.Attributes.ServiceRegistry == provider.Kubernetes` part is problematic, at least wrt to current code.

Use your VS example but replace `helloworld` with something that is NOT a Service. You will get only `fake-name.ns.svc.cluster.local`.

It seems like if we expand it, we should always fully expand it. But we probably shouldn't expand shortname at all in this case since we don't know its a k8s service...

Reply:
Yes, it is documented shortnames can be used only for k8s.